src/extensions/scratch3_video_sensing/view.js

```diff
@@ -0,0 +1,328 @@
+const WIDTH = 480;
+const HEIGHT = 360;
+const WINSIZE = 8;
+const AMOUNT_SCALE = 100;
+const THRESHOLD = 10;
+
+const OUTPUT = {
+    INPUT: -1,
+    XYT: 0,
+    XYT_CELL: 1,
+    XY: 2,
+    XY_CELL: 3,
+    T: 4,
+    T_CELL: 5,
+    C: 6,
+    AB: 7,
+    UV: 8
+};
+
+class VideoMotionView {
+    constructor (motion, output = OUTPUT.XYT) {
+        this.motion = motion;
+
+        const canvas = this.canvas = document.createElement('canvas');
+        canvas.width = WIDTH;
+        canvas.height = HEIGHT;
+        this.context = canvas.getContext('2d');
+
+        this.output = output;
+        this.buffer = new Uint32Array(WIDTH * HEIGHT);
+    }
+
+    static get OUTPUT () {
+        return OUTPUT;
+    }
+
+    draw () {
```

Comment:
There's a lot of code duplicated here... is that intentional? Could they share code, or is one of the copies obsolete?

Reply:
Its duplicated intentionally. A lot of it was finding what needed to be copied to get the output renders to help understand the algorithm output in real time. The gradX, gradY, gradT calculations could be deduped. Some of the copies are out of date in minor ways.